elfpy/data/contract_interface.py

```diff
@@ -56,17 +67,64 @@ class HyperdriveAddressesJson:
     mock_hyperdrive_math: str = attr.ib()
 
 
-def get_account_balance_for_contract(funding_contract: Contract, account_address: str) -> int:
-    """Return the balance of the account"""
-    return funding_contract.functions.balanceOf(account_address).call()
+def initialize_web3_with_http_provider(ethereum_node: URI | str, request_kwargs: dict | None = None) -> Web3:
+    """Initialize a Web3 instance using an HTTP provider and inject a geth Proof of Authority (poa) middleware.
+
+    Arguments
+    ---------
+    ethereum_node: URI | str
+        Address of the http provider
+    request_kwargs: dict
+        The HTTPProvider uses the python requests library for making requests.
+        If you would like to modify how requests are made,
+        you can use the request_kwargs to do so.
+    """
+    if request_kwargs is None:
+        request_kwargs = {}
+    provider = Web3.HTTPProvider(ethereum_node, request_kwargs)
+    web3 = Web3(provider)
+    web3.middleware_onion.inject(geth_poa.geth_poa_middleware, layer=0)
+    return web3
+
+
+def set_anvil_account_balace(web3: Web3, account_address: str, amount_wei: int) -> RPCResponse:
+    """Set an the account using the web3 provider
+
+    Arguments
+    ---------
+    amount_wei : int
+        amount_wei to fund, in wei
+    """
+    if not web3.is_checksum_address(account_address):
+        raise ValueError(f"argument {account_address=} must be a checksum address")
+    params = [account_address, hex(amount_wei)]  # account, amount
+    rpc_response = web3.provider.make_request(method=RPCEndpoint("anvil_setBalance"), params=params)
+    return rpc_response
 
 
-def fund_account(funding_contract: Contract, account_address: str, amount: int) -> HexBytes:
-    """Add funds to the account"""
-    tx_receipt = funding_contract.functions.mint(account_address, amount).transact()
+def mint_tokens(token_contract: Contract, account_address: str, amount_wei: int) -> HexBytes:
+    """Add funds to the account
+
+    Arguments
+    ---------
+    amount_wei : int
+        amount_wei to fund, in wei
+    """
+    tx_receipt = token_contract.functions.mint(account_address, amount_wei).transact()
     return tx_receipt
 
 
+def get_account_balance_from_provider(web3: Web3, account_address: str) -> int | None:
+    """Get the balance for an account deployed on the web3 provider"""
+    if not web3.is_checksum_address(account_address):
+        raise ValueError(f"argument {account_address=} must be a checksum address")
+    rpc_response = web3.provider.make_request(method=RPCEndpoint("eth_getBalance"), params=[account_address, "latest"])
+    result = rpc_response.get("result")
+    if result is not None:
+        return int(result, 16)
```

Comment:
Unclear what this line is doing, you're casting a result to an integer, or the default value is 16? Can you comment this line?